From 55fc0727534575897a866f4b89295dfc38652434 Mon Sep 17 00:00:00 2001
From: Ulf Hermann <ulf.hermann@qt.io>
Date: Tue, 22 Feb 2022 12:31:08 +0100
Subject: [PATCH 32/58] Reduce memory leakage

We need to clean up the event queue when we're done.

Change-Id: I13a9eb77e978f4eab227a3a28dab8ebc8de94405
Reviewed-by: David Edmundson <davidedmundson@kde.org>
(cherry picked from commit 1264e5f565d8fb7ac200e4b00531ab876922458f)
---
 src/client/qwaylanddisplay.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/client/qwaylanddisplay.cpp b/src/client/qwaylanddisplay.cpp
index 41a44e4b..622ffe41 100644
--- a/src/client/qwaylanddisplay.cpp
+++ b/src/client/qwaylanddisplay.cpp
@@ -379,6 +379,9 @@ QWaylandDisplay::~QWaylandDisplay(void)
 #endif
     if (mDisplay)
         wl_display_disconnect(mDisplay);
+
+    if (m_frameEventQueue)
+        wl_event_queue_destroy(m_frameEventQueue);
 }
 
 // Steps which is called just after constructor. This separates registry_global() out of the constructor
-- 
2.39.1

